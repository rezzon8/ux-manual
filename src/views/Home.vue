<template>
  <v-container class="home">
    <v-row class="text-center">
      <v-col class="mb-4">
        <v-icon size="42" class="mr-3">mdi-book-open-page-variant</v-icon>
        <h1 class="display-2 font-weight-bold mb-3 mt-3">
          The UX Manual
        </h1>

        <p class="subheading font-weight-regular">
          A reference guide and resouce for the processes <br />
          of which UX strategy is comprised
        </p>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <UXProcessSection
          v-for="item in $store.getters.getUXProcesses"
          :key="item.id"
          :title="item.title"
          :copy="item.copy"
          :color="item.color"
          :process_steps="$store.getters.getProcessSteps(item.id)"
        />
        <Dialog :dialog="displayDialog"></Dialog>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
// @ is an alias to /src
import UXProcessSection from "@/components/UXProcessSection.vue";
import Dialog from "@/components/Dialog.vue";

export default {
  name: "UX_manual",
  components: {
    UXProcessSection,
    Dialog
  },
  computed: {
    isDataLoaded() {
      return this.$store.state.is_data_loaded;
    },
    displayDialog() {
      return true;
    }
  },
  beforeCreate() {
    this.$store
      .dispatch("GET_DATA")
      .then(() => {
        console.log("asdas");
      })
      .catch(e => {
        console.log(e.message);
      });
  }
};
</script>
