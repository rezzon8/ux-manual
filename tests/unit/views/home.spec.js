import { createLocalVue, mount } from "@vue/test-utils";
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
import VueRouter from "vue-router";
import Vuelidate from "vuelidate";
import Home from "@/views/Home.vue";

Vue.use(Vuelidate);
Vue.use(Vuetify); // added to prevent vuetify lint errors

const localVue = createLocalVue();
const vuetify = new Vuetify();

let wrapper;
let router;
let store;
let state;
let getters;
let actions;
let mutations;

localVue.use(VueRouter);
localVue.use(Vuex);

router = new VueRouter();

// fixes: [Vuetify] Unable to locate target [data-app]
document.body.setAttribute("data-app", true);

const factory = () => {
  return mount(Home, {
    propsData: {},
    computed: {
      getProcessStep() {
        return jest.fn(() => {
          return { title: "asda", copy: { introduction: "copy" } };
        });
      }
    },
    data: () => ({
      color: "blue",
      dialog: {
        title: "UX Process Step",
        copy: {},
        visible: false
      }
    }),
    mocks: {},
    stubs: {},
    store,
    vuetify,
    router,
    localVue
  });
};

describe("App.vue", () => {
  beforeEach(() => {
    state = {
      process_steps: [
        {
          id: 1,
          title: "Test Title",
          copy: "Test copy goes here..."
        }
      ]
    };
    getters = {
      userIsAuthenticated: jest.fn(() => false)
    };
    actions = {
      GET_DATA: jest.fn(),
      LOGIN: jest.fn(),
      SIGN_UP: jest.fn(),
      LOGOUT: jest.fn()
    };
    mutations = {
      someMutation: jest.fn()
    };
    store = new Vuex.Store({
      state,
      getters,
      actions,
      mutations
    });
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });
  it("should render", () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it("should define a closeDialog method", () => {
    expect(wrapper.vm.closeDialog).toBeDefined();
  });
  it("should define a openDialog method", () => {
    expect(wrapper.vm.openDialog).toBeDefined();
  });
  it("should define a setSelectedStep method", () => {
    expect(wrapper.vm.setSelectedStep).toBeDefined();
  });
  it("should hide the dialog", () => {
    wrapper.vm.dialog.visible = true;
    wrapper.vm.closeDialog();
    expect(wrapper.vm.dialog.visible).toBe(false);
  });
  it("should fail to run method if dialog.visible is false", () => {
    // NOTE: testing watcher
    wrapper.vm.$options.watch["dialog.visible"].call(wrapper.vm, false);
    wrapper.vm.closeDialog();
    expect(wrapper.vm.dialog.visible).toBe(false);
  });
  it("should show the dialog", () => {
    wrapper.vm.dialog.visible = false;
    wrapper.vm.openDialog({ id: 1, color: "blue" });
    expect(wrapper.vm.dialog.visible).toBe(true);
  });
});
