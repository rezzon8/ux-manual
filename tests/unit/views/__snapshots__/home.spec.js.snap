// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render 1`] = `
<div class="container home">
  <div class="row text-center">
    <div class="mb-4 col">
      <h1 class="display-2 font-weight-bold mb-3 mt-3 animate__animated animate__tada animate__delay-2s"><i aria-hidden="true" class="v-icon notranslate mb-3 mdi mdi-book-open-page-variant theme--light" style="font-size: 62px;"></i> <br>
        The UX Manual
      </h1>
      <p class="subheading font-weight-regular">
        A reference guide and resource for the processes of which UX strategy
        is comprised. <br>Please note that this app is
        <strong>purely for demonstration purposes only!</strong> and forms
        part of my portfolio.
      </p>
      <p class="subheading font-weight-regular">
        You are currently not logged in. Feel free to signup to reveal all the
        UX process steps below.
      </p>
    </div>
  </div>
  <div class="row">
    <div class="col">
      <div class="text-center">
        <div role="dialog" class="v-dialog__container">
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
